### PR TITLE
[extractor/kick] Add extractor

### LINKED ISSUE #5722: kick.com
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

United States

### Example URLs

- Livestream https://kick.com/kmack710 (might be offline)
- Any URL from the live channels page on https://kick.com/categories in case above is offline

### Provide a description that is worded well enough to be understood

The site appears to use all AWS services, live-video.net is the domain for the m3u8 playlist. Streams work without logging in.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://kick.com/kmack710
[debug] Command-line config: ['-vU', 'https://kick.com/kmack710']
[debug] Portable config "C:\Users\User\scoop\apps\yt-dlp\current\yt-dlp.conf": []
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22621-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109235-g1009396953-20221127 (setts), ffprobe N-109235-g1009396953-20221127
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://kick.com/kmack710
[generic] kmack710: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] kmack710: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://kick.com/kmack710
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1485, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1561, in __extract_info
  File "yt_dlp\extractor\common.py", line 674, in extract
  File "yt_dlp\extractor\generic.py", line 2721, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://kick.com/kmack710
```

### PR DESCRIPTION
**Pull request by bashonly**
kick.com is a twitch.tv clone currently in "beta"

Closes #5722

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by bashonly** [comment]
Update -- these pages actually exist now:
https://kick.com/privacy-policy
https://kick.com/terms-of-service

It appears the site does actually care about being legitimate

File: yt_dlp/extractor/kick.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        if not self._API_HEADERS:
+            self._request_webpage(HEADRequest('https://kick.com/'), display_id, 'Setting up session')
+            xsrf_token = self._get_cookies('https://kick.com/').get('XSRF-TOKEN')
+            if xsrf_token:
+                self._API_HEADERS.update({
+                    'Authorization': f'Bearer {xsrf_token.value}',
+                    'X-XSRF-TOKEN': xsrf_token.value,
+                })
+            else:
+                self.write_debug('kick.com did not set XSRF-TOKEN cookie')
```
Comment (pukkandan) [inline yt_dlp/extractor/kick.py:28]: put this in `_real_initialize`

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/kick.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    def _real_initialize(self):
+        self._request_webpage(HEADRequest('https://kick.com/'), None, 'Setting up session')
+        xsrf_token = self._get_cookies('https://kick.com/').get('XSRF-TOKEN')
+        if xsrf_token:
+            self._API_HEADERS.update({
+                'Authorization': f'Bearer {xsrf_token.value}',
+                'X-XSRF-TOKEN': xsrf_token.value,
+            })
+        else:
+            self.write_debug('kick.com did not set XSRF-TOKEN cookie')
```
Comment (pukkandan) [inline yt_dlp/extractor/kick.py:27]:
```suggestion
    def _real_initialize(self):
        self._request_webpage(HEADRequest('https://kick.com/'), None, 'Setting up session')
        xsrf_token = self._get_cookies('https://kick.com/').get('XSRF-TOKEN')
        if not xsrf_token:
            self.write_debug('kick.com did not set XSRF-TOKEN cookie')
        KickBaseIE._API_HEADERS = {
            'Authorization': f'Bearer {xsrf_token.value}',
            'X-XSRF-TOKEN': xsrf_token.value,
        } if xsrf_token else {}
```

**Review by pukkandan — APPROVED** [review]

**Comment by camjac251** [comment]
Is this working still. I tried it out today with version `2023.01.02` and get this error
```
yt-dlp -U

Latest version: 2023.01.02, Current version: 2023.01.02
yt-dlp is up to date (2023.01.02)

yt-dlp https://kick.com/kmack710

[Kick] Setting up session
ERROR: [Kick] kmack710: Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

```

**Comment by bashonly** [comment]
@camjac251 I can't reproduce; it is still working for me. Can you open a site bug issue with a verbose log (the output of `yt-dlp -vU https://kick.com/kmack710`)?

**Comment by dragu** [comment]
@bashonly 

```
$ ./yt-dlp https://kick.com/suspendas -vU
[debug] Command-line config: ['https://kick.com/suspendas', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.06 [6becd2508] (zip)
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-5.10.0-21-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.31)
[debug] exe versions: ffmpeg 4.3.5-0, ffprobe 4.3.5-0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.9.7, certifi-2020.06.20, pyxattr-0.7.2, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1764 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.01.06, Current version: 2023.01.06
yt-dlp is up to date (2023.01.06)
[Kick] Setting up session
ERROR: [Kick] suspendas: Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
File "./yt-dlp/yt_dlp/extractor/common.py", line 679, in extract
self.initialize()
File "./yt-dlp/yt_dlp/extractor/common.py", line 588, in initialize
self._real_initialize()
File "./yt-dlp/yt_dlp/extractor/kick.py", line 17, in _real_initialize
self._request_webpage(HEADRequest('https://kick.com/'), None, 'Setting up session')
File "./yt-dlp/yt_dlp/extractor/common.py", line 827, in _request_webpage
raise ExtractorError(errmsg, cause=err)

File "./yt-dlp/yt_dlp/extractor/common.py", line 809, in _request_webpage
return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
File "./yt-dlp/yt_dlp/YoutubeDL.py", line 3730, in urlopen
return self._opener.open(req, timeout=self._socket_timeout)
File "/usr/lib/python3.9/urllib/request.py", line 523, in open
response = meth(req, response)
File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
response = self.parent.error(
File "/usr/lib/python3.9/urllib/request.py", line 561, in error
return self._call_chain(*args)
File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
result = func(*args)
File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

I contacted kick several times, they claimed to have added my IP address to the whitelist, and yet I also get a 403 error.

**Comment by bashonly** [comment]
> I contacted kick several times, they claimed to have added my IP address to the whitelist, and yet I also get a 403 error.

Does kick.com load in your web browser?

**Comment by dragu** [comment]
@bashonly I am running yt-dlp on the server, so I only checked in a text browser without JS, which is running cloudflare challange without success.

wget https://kick.com immediately gets 403

**Pull request merged**